client/pages/trip-todo.jsx

```diff
@@ -37,6 +38,19 @@ export default class TripTodo extends React.Component {
       .catch(err => console.error(err));
   }
 
+  updateCompleted(todoId) {
+    const newTodos = Array.from(this.state.todos);
+    let index;
+    for (let i = 0; i < newTodos.length; i++) {
+      if (newTodos[i].todoId === todoId) {
+        index = i;
+      }
+    }
+    const completedStatus = newTodos[index].completed;
+    newTodos[index].completed = !completedStatus;
```

Comment:
Careful, objects are reference data and we don't want to do assignment to their properties. You have copied the egg carton, but not the eggs inside. You can copy an object with either [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Cloning_an_object) or [spread properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Syntax).